contracts/src/starknet/rewaave/token_bridge.cairo

```diff
@@ -221,11 +229,29 @@ func handle_deposit{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_chec
     end
     let amount = Uint256(low=amount_low, high=amount_high)
 
-    assert_not_zero(l2_token)
+    assert_not_zero(l2_token_address)
 
     # Call mint on l2_token contract.
-    IERC20.mint(contract_address=l2_token, recipient=l2_recipient, amount=amount)
-    deposit_handled.emit(l2_token, l2_recipient, amount)
 
+    IERC20.mint(contract_address=l2_token_address, recipient=l2_recipient, amount=amount)
+    deposit_handled.emit(l2_token_address, l2_recipient, amount)
+    return ()
+end
+
+@external
+func mint_rewards{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    recipient : felt, amount : Uint256
+):
+    # get the address of the ETHStaticAToken
+    let (l2_token) = get_caller_address()
+    # Verify that it's a valid token by checking for its counterpart on l1
+    let (l1_token) = l2_token_to_l1_token.read(l2_token)
```

Comment:
We do this twice, can we break it out into a library function? `is_token(l2_token)`